dot/parachain/availability-distribution/availability_distribution_test.go

```diff
@@ -248,3 +251,176 @@ func TestHandlePoVFetchingRequest(t *testing.T) {
 		wg.Wait()
 	})
 }
+
+func Test_getBlockAncestorsInSameSession(t *testing.T) {
+	var (
+		ctrl           *gomock.Controller
+		netMock        *MockNetwork
+		blockStateMock *MockBlockState
+		runtimeMock    *MockInstance
+		overseerCh     chan any
+		ad             *AvailabilityDistribution
+	)
+
+	setup := func(t *testing.T) {
+		ctrl = gomock.NewController(t)
+		netMock = NewMockNetwork(ctrl)
+		blockStateMock = NewMockBlockState(ctrl)
+		runtimeMock = NewMockInstance(ctrl)
+
+		netMock.EXPECT().RegisterRequestHandler(protocol.ID("req_chunk/2"), gomock.Any())
+		netMock.EXPECT().RegisterRequestHandler(protocol.ID("req_pov/1"), gomock.Any())
+
+		blockStateMock.EXPECT().
+			GetRuntime(gomock.AssignableToTypeOf(common.Hash{})).
+			MaxTimes(leafAncestryLenWithinSession+1).
+			Return(runtimeMock, nil)
+
+		runtimeMock.EXPECT().Stop().MaxTimes(leafAncestryLenWithinSession + 1)
+
+		overseerCh = make(chan any)
+		ad = NewAvailabilityDistribution(overseerCh, netMock, blockStateMock)
+	}
+
+	t.Run("no_header_for_leaf_hash", func(t *testing.T) {
```

Comment:
Can we organize the test cases using a struct to improve readability and maintainability? This will help in clearly defining test inputs and expectations in a structured manner.